core/src/main/java/org/apache/druid/data/input/impl/FirehoseToInputSourceReaderAdaptor.java

```
@@ -48,7 +49,7 @@ public FirehoseToInputSourceReaderAdaptor(
   }
 
   @Override
-  public CloseableIterator<InputRow> read() throws IOException
+  public CloseableIterator<InputRow> read(InputStats inputStats) throws IOException
```

Maybe throw UnsupportedOpException here?